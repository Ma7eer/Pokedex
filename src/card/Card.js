import React from 'react';

const Card = (props) => {
  return (
      <div className="card px-5 mx-4 text-white bg-dark mb-3" style={{width: '15rem'}}>
        <img className="card-img-top" alt="pokemon sprite" src={props.img} />
        <div className="card-body">
          <div>{props.id}</div>
          <div>{props.name}</div>
        </div>
      </div>
  )
}

export default Card