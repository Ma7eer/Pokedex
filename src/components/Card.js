import React from 'react';

// insert the array of items created into our local storage
// localStorage.setItem('items', JSON.stringify(itemsArray));
// const data = JSON.parse(localStorage.getItem('items'));

const handleClick = (pokemonArray) => {
  /* create an array that will either be an empty array or
  will be all the items listed in our local storage from a previous session */
  let localStorageArray = localStorage.getItem('data') ? JSON.parse(localStorage.getItem('data')) : [];

  localStorageArray.push.apply(localStorageArray, pokemonArray); // pushes data received from our card into our itemsArray

  localStorage.setItem('data', JSON.stringify(localStorageArray)); // adds that data from card into the local storage
  console.log(localStorage);
}

const Card = (props) => {

  return (
    <div style={{margin: '1em'}} className="bg-dark p-4">
      <img className="hvr-bob" width="180" src={props.spriteImgUrl} alt=""/>
      <div style={{display: 'inline-block', marginLeft: 10}}>
        <div style={{fontSize: '1.25em', fontWeight: 'bold'}}>
        {props.name}
        </div>
        <div>Pokedex ID: {props.id}</div>
        <div>Type: {props.type}</div>
        <button onClick={() => handleClick([props.spriteImgUrl, props.name, props.id, props.type])} type="button" className="btn btn-primary mt-4">Add to list</button>
      </div>
      </div>
  )
}

export default Card