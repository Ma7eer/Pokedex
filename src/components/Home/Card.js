import React from 'react';

 const Card = (props) => {
  const pokemonType = (type) => {
      const typeObject = {
        'normal': '#a8a878', 'fire': '#f08030',
        'fighting': '#c03028', 'water': '#6890f0',
        'flying': '#a890f0', 'grass': '#78c850',
        'poison': '#a040a0', 'electric': '#f8d030',
        'ground': '#e0c068', 'psychic': '#f85888',
        'rock': '#b8a038', 'ice': '#98d8d8',
        'bug': '#a8b820', 'dragon': '#7038f8',
        'ghost': '#705898', 'dark': '#705848',
        'steel': '#b8b8d0', 'fairy': '#ee99ac'
      }
      return typeObject[type];
    }

  // store pokemon data to local Storage
  const handleClick = (pokemonArray) => {
    let myMap = new Map();
    myMap.set(pokemonArray[0], [pokemonArray[1], pokemonArray[2], pokemonArray[3]])
    myMap.forEach((val, key) => {
      localStorage.setItem(key, val);
      console.log(localStorage.getItem(1));
    });
  }

    return (
    <div style={{margin: '1em'}} className="bg-dark p-4">
      <img className="hvr-bob" width="180" src={props.spriteImgUrl} alt=""/>
      <div style={{display: 'inline-block', marginLeft: 10}}>
        <div style={{fontSize: '1.25em', fontWeight: 'bold'}}>
        {props.name}
        </div>
        <div>Pokedex ID: {props.id}</div>
        <div style={pokemonType([props.type]) ? {background: pokemonType([props.type])} : {background: '#68a090'}}
        className="rounded d-flex justify-content-center mt-2">
        Type: {props.type}</div>
        <button onClick={() => handleClick([props.id, props.spriteImgUrl, props.name, props.type])} type="button" className="btn btn-primary mt-3">Add to list</button>
      </div>
      </div>
  )
}

export default Card;