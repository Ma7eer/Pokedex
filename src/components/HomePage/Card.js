import React from 'react';
import { pokemonType } from './pokemonType';
import PropTypes from 'prop-types';
import toastr from 'toastr';
import CardButton from './CardButton';

const Card = (props) => {
   // props
  const { spriteImgUrl, name, id, type } = props;

  // store pokemon data to local Storage
  const handleClick = (pokemonArray) => {
    let myMap = new Map();
    myMap.set(pokemonArray[0], [pokemonArray[1], pokemonArray[2], pokemonArray[3]])
    myMap.forEach((val, key) => {
      localStorage.setItem(key, val);
      console.log(localStorage.getItem(1));
    });
    toastr.info('pokemon added to your list :]');
  }

  return (
    <div style={{margin: '1em'}} className="bg-dark p-4">
      <img className="hvr-bob" width="180" src={spriteImgUrl} alt=""/>
      <div style={{display: 'inline-block', marginLeft: 10}}>
        <div style={{fontSize: '1.25em', fontWeight: 'bold'}}>
        {name}
        </div>
        <div>Pokedex ID: {id}</div>
        <div style={pokemonType([type]) ? {background: pokemonType([type])} : {background: '#68a090'}}
        className="rounded d-flex justify-content-center mt-2">
        Type: {type}</div>
        <CardButton onClick={() => handleClick([id, spriteImgUrl, name, type])}/>
      </div>
    </div>
  )
}

Card.propTypes = {
  spriteImgUrl: PropTypes.string,
  name: PropTypes.string,
  id: PropTypes.number,
  type: PropTypes.string
}

export default Card;