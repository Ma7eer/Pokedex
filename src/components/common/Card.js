import React from 'react';
import { pokemonType } from '../HomePage/pokemonType';
import PropTypes from 'prop-types';
import toastr from 'toastr';
import CardButton from './CardButton';

const Card = (props) => {
   // props
  const { spriteImgUrl, name, id, type } = props;

  // store pokemon data to local Storage
  const handleAddClick = (pokemonArray) => {
    let myMap = new Map();
    myMap.set(pokemonArray[0], [pokemonArray[1], pokemonArray[2], pokemonArray[3]])
    myMap.forEach((val, key) => {
      if(localStorage.getItem(key)) {
        toastr.warning(`${name} already added to your list!`);
      } else {
      localStorage.setItem(key, val);
      // console.log(localStorage.getItem(1));
      toastr.info(`${name} added to your list :]`);
    }
    });
  }

  const handleDeleteClick = (key) => {
    console.log('deleted NOT');
    localStorage.removeItem(key);
  }

  return (
    <div style={{margin: '1em'}} className="bg-dark p-4">
      <img className="hvr-bob" width="180" src={spriteImgUrl} alt=""/>
      <div style={{display: 'inline-block', marginLeft: 10}}>
        <div style={{fontSize: '1.25em', fontWeight: 'bold'}}>
        {name}
        </div>
        <div>Pokedex ID: {id}</div>
        <div style={pokemonType([type]) ? {background: pokemonType([type])} : {background: '#68a090'}}
        className="rounded d-flex justify-content-center mt-2">
        Type: {type}</div>
        <CardButton onClick={() => handleAddClick([id, spriteImgUrl, name, type])} onDelete={() => {handleDeleteClick(id)}} homePath={props.homePath} id={id}/>
      </div>
    </div>
  )
}

Card.propTypes = {
  spriteImgUrl: PropTypes.string,
  name: PropTypes.string,
  id: PropTypes.number,
  type: PropTypes.string
}

export default Card;