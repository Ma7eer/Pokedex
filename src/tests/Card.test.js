import React from 'react';
import renderer from 'react-test-renderer';

import Card from '../components/Home/Card';

// running some snapshot tests on out Card Component
it('renders without breaking if no data is fed to it', () => {
  const tree = renderer.create(<Card />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders without breaking if dummy data is fed to it', () => {
  const tree = renderer.create(<Card
    type="electric"
    id="7"
    name="pikachu"
    spriteImgUrl="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/4.png"/>).toJSON();
  expect(tree).toMatchSnapshot();
});