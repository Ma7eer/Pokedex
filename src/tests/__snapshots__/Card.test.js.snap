// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without breaking if dummy data is fed to it 1`] = `
<div
  className="bg-dark p-4"
  style={
    Object {
      "margin": "1em",
    }
  }
>
  <img
    alt=""
    className="hvr-bob"
    src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/4.png"
    width="180"
  />
  <div
    style={
      Object {
        "display": "inline-block",
        "marginLeft": 10,
      }
    }
  >
    <div
      style={
        Object {
          "fontSize": "1.25em",
          "fontWeight": "bold",
        }
      }
    >
      pikachu
    </div>
    <div>
      Pokedex ID: 
      7
    </div>
    <div
      className="rounded d-flex justify-content-center mt-2"
      style={
        Object {
          "background": "#f8d030",
        }
      }
    >
      Type: 
      electric
    </div>
    <button
      className="btn btn-primary mt-3"
      onClick={[Function]}
      type="button"
    >
      Add to list
    </button>
  </div>
</div>
`;

exports[`renders without breaking if no data is fed to it 1`] = `
<div
  className="bg-dark p-4"
  style={
    Object {
      "margin": "1em",
    }
  }
>
  <img
    alt=""
    className="hvr-bob"
    src={undefined}
    width="180"
  />
  <div
    style={
      Object {
        "display": "inline-block",
        "marginLeft": 10,
      }
    }
  >
    <div
      style={
        Object {
          "fontSize": "1.25em",
          "fontWeight": "bold",
        }
      }
    />
    <div>
      Pokedex ID: 
    </div>
    <div
      className="rounded d-flex justify-content-center mt-2"
      style={
        Object {
          "background": "#68a090",
        }
      }
    >
      Type: 
    </div>
    <button
      className="btn btn-primary mt-3"
      onClick={[Function]}
      type="button"
    >
      Add to list
    </button>
  </div>
</div>
`;
